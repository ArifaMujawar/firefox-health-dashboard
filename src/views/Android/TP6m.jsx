/* eslint-disable react/no-array-index-key */
import React from 'react';
import { withStyles } from '@material-ui/core';
import PropTypes from 'prop-types';
import Grid from '@material-ui/core/Grid';
import { round } from '../../vendor/math';
import { exists, missing } from '../../vendor/utils';
import { selectFrom } from '../../vendor/vectors';
import {
  PLATFORMS,
  TP6_COMBOS,
  TP6_TESTS,
  TP6M_SITES,
} from '../../quantum/config';
import { withNavigation } from '../../vendor/utils/navigation';
import Picker from '../../vendor/utils/navigation/Picker';
import DashboardPage from '../../components/DashboardPage';
import PerfherderGraphContainer from '../../containers/PerfherderGraphContainer';
import ChartJSWrapper from '../../vendor/chartJs/ChartJsWrapper';
import { g5Reference, TARGET_NAME } from '../../config/mobileG5';
import { pullAggregate } from '../../components/TP6mAggregate';
import Section from '../../components/Section';

const styles = {
  body: {
    backgroundColor: 'white',
  },
  chart: {
    justifyContent: 'center',
    padding: '1rem',
  },
};

class TP6M extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  async componentDidMount() {
    const { test, platform } = this.props;
    const timeRange = 'today-6week';
    const tests = selectFrom(TP6_TESTS).where({ test });
    const testMode = tests.select('mode').first();
    const sites = TP6M_SITES.filter(({ mode }) =>
      mode.includes(testMode)
    ).materialize();
    const aggregate = await pullAggregate({
      condition: {
        or: TP6_COMBOS.where({ test, platform }).select('seriesConfig'),
      },
      sites,
      tests,
      platforms: selectFrom(PLATFORMS).where({ platform }),
      timeRange,
    });
    const referenceValue = aggregate.where({ test, platform }).ref.getValue();

    if (missing(referenceValue)) {
      // THERE IS NO GEOMEAN TO CALCULATE
      this.setState({ summaryData: null, test, platform });

      return;
    }

    const summaryData = {
      datasets: aggregate
        .where({ test, platform })
        .along('platform') // dummy (only one)
        .map(({ result }) => ({
          label: platform,
          data: result
            .along('pushDate')
            .map(point => ({
              x: point.pushDate.getValue(),
              y: point.getValue(),
            }))
            .toArray(),
        }))
        .append(
          exists(referenceValue) && {
            label: TARGET_NAME,
            style: {
              type: 'line',
              backgroundColor: 'gray',
              borderColor: 'gray',
              fill: false,
              pointRadius: '0',
              pointHoverBackgroundColor: 'gray',
              lineTension: 0,
            },
            data: aggregate
              .where({ test, platform })
              .along('pushDate')
              .map(({ pushDate, ref }) => ({
                x: pushDate.getValue(),
                y: ref.getValue(),
              }))
              .toArray(),
          }
        )
        .toArray(),
    };

    this.setState({ summaryData, test, platform });
  }

  async componentDidUpdate(prevProps) {
    if (
      this.props.test === prevProps.test &&
      this.props.platform === prevProps.platform
    )
      return;

    this.componentDidMount();
  }

  render() {
    const { classes, navigation, test, platform } = this.props;
    let { summaryData } = this.state;

    if (test !== this.state.test || platform !== this.state.platform) {
      summaryData = null;
    }

    const subtitle = selectFrom(TP6_TESTS)
      .where({ test })
      .first().label;

    return (
      <div className={classes.body}>
        <DashboardPage key={subtitle} title="TP6 Mobile" subtitle={subtitle}>
          <Section title="Details by site">
            <Grid container spacing={24}>
              <Grid item xs={6} className={classes.chart}>
                {navigation}
              </Grid>
              <Grid item xs={6} className={classes.chart}>
                {summaryData && (
                  <ChartJSWrapper
                    title={`Geomean of ${subtitle}`}
                    data={summaryData}
                    height={200}
                  />
                )}
              </Grid>

              {selectFrom(TP6_COMBOS)
                .where({
                  browser: 'geckoview',
                  platform,
                  test,
                })
                .groupBy('site')
                .map((series, site) => (
                  <Grid
                    item
                    xs={6}
                    key={`page_${site}_${test}_${platform}`}
                    className={classes.chart}>
                    <PerfherderGraphContainer
                      title={site}
                      reference={(() => {
                        const value = round(
                          g5Reference
                            .where({ test, platform, site })
                            .getValue(),
                          { places: 2 }
                        );

                        return {
                          label: `Target (${value})`,
                          value,
                        };
                      })()}
                      series={selectFrom(series)
                        .sortBy(['ordering'])
                        .map(s => ({
                          label: s.browser,
                          seriesConfig: s.seriesConfig,
                        }))
                        .toArray()}
                    />
                  </Grid>
                ))}
            </Grid>
          </Section>
        </DashboardPage>
      </div>
    );
  }
}

TP6M.propTypes = {
  classes: PropTypes.shape({}).isRequired,
  location: PropTypes.shape({
    search: PropTypes.string.isRequired,
  }).isRequired,
  test: PropTypes.string.isRequired,
  platform: PropTypes.string.isRequired,
};

const nav = [
  {
    type: Picker,
    id: 'test',
    label: 'Test',
    defaultValue: 'warm-loadtime',
    options: selectFrom(TP6_TESTS)
      .select({ id: 'test', label: 'label' })
      .toArray(),
  },

  {
    type: Picker,
    id: 'platform',
    label: 'Platform',
    defaultValue: 'android-g5',
    options: selectFrom(PLATFORMS)
      .where({ browser: 'geckoview' })
      .select({ id: 'platform', label: 'label' })
      .toArray(),
  },
];

export default withNavigation(nav)(withStyles(styles)(TP6M));
