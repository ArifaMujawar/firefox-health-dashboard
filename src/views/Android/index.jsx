/* eslint-disable react/no-array-index-key */
import React, { Component } from 'react';
import Grid from '@material-ui/core/Grid';
import DashboardPage from '../../components/DashboardPage';
import Section from '../../components/Section';
import BugzillaUrlContainer from '../../containers/BugzillaUrlContainer';
import BugzillaGraph from '../../containers/BugzillaGraph';
import NimbledroidSection from '../../containers/NimbledroidSection';
import PerfherderGraphContainer from '../../containers/PerfherderGraphContainer';
import RedashContainer from '../../containers/RedashContainer';
import CONFIG from '../../utils/nimbledroid/config';
import { TP6mAggregate } from '../../components/TP6mAggregate';

class Android extends Component {
  render() {
    const nimbledroidSubTitle = `${
      CONFIG.packageIdLabels[CONFIG.baseProduct]
    } vs ${CONFIG.packageIdLabels[CONFIG.compareProduct]}`;

    return (
      <DashboardPage title="Android" subtitle="Release criteria">
        <div>
          <Grid container spacing={24}>
            <Grid item xs={6} key="bugzilla">
              <Section title="Bugzilla">
                <BugzillaUrlContainer
                  includeBugCount
                  queries={[
                    {
                      text: 'All GV Firefox Preview MVP bugs',
                      parameters: {
                        resolution: '---',
                        status_whiteboard: '[geckoview:fenix:m',
                        status_whiteboard_type: 'substring',
                      },
                    },
                  ]}
                />
                <BugzillaGraph
                  queries={[
                    {
                      label: 'GV M3 bugs',
                      parameters: {
                        resolution: ['---', 'FIXED'],
                        whiteboard: [
                          '[geckoview:fenix:m2]',
                          '[geckoview:fenix:m3]',
                        ],
                        status_whiteboard_type: 'anywordssubstr',
                      },
                    },
                    {
                      label: 'GV M4 bugs',
                      parameters: {
                        resolution: ['---', 'FIXED'],
                        whiteboard: '[geckoview:fenix:m4]',
                      },
                    },
                    {
                      label: 'GV M5 bugs',
                      parameters: {
                        resolution: ['---', 'FIXED'],
                        whiteboard: '[geckoview:fenix:m5]',
                      },
                    },
                  ]}
                  startDate="2019-02-01"
                  title="Firefox Preview Bugs"
                />
              </Section>
            </Grid>

            <Grid item xs={6} key="nimbledroid">
              <Section title="Nimbledroid" subtitle={nimbledroidSubTitle}>
                <NimbledroidSection configuration={{ ...CONFIG }} />
              </Section>
            </Grid>
          </Grid>
        </div>
        <Section
          title="Raptor (TP6m)"
          more="/android/tp6m?test=cold-loadtime&platform=android-p2-aarch64">
          <TP6mAggregate />
        </Section>
        <Section title="Telemetry">
          <Grid container spacing={24}>
            <Grid item xs={6}>
              <RedashContainer
                title="Total content page load time (no 95th)"
                redashDataUrl="https://sql.telemetry.mozilla.org/api/queries/59395/results.json?api_key=2L0YcuUULtECr9bfew9OAEgtC50G4Ri8NCSPLR5F"
                redashQueryUrl="https://sql.telemetry.mozilla.org/queries/59395"
              />
            </Grid>
            <Grid item xs={6}>
              <RedashContainer
                title="Total content page load time"
                redashDataUrl="https://sql.telemetry.mozilla.org/api/queries/59397/results.json?api_key=u9eculhXgxqgsluxYGxfXaWQ6g7KCXioEvfwjK83"
                redashQueryUrl="https://sql.telemetry.mozilla.org/queries/59397"
              />
            </Grid>
          </Grid>
        </Section>
        <Section
          title="Perfherder"
          subtitle="Lower in the graph is better regardless if it is a score or execution time (read the Y label)">
          <Grid container spacing={24}>
            <Grid item xs={6}>
              <PerfherderGraphContainer
                title="Speedometer"
                series={[
                  {
                    label: 'Moto G5 (arm7)',
                    seriesConfig: {
                      and: [
                        { missing: 'test' },
                        {
                          prefix: { platform: 'android-hw-g5-7-0-arm7-api-16' },
                        },
                        {
                          eq: {
                            framework: 10,
                            repo: 'mozilla-central',
                            suite: 'raptor-speedometer-geckoview',
                          },
                        },
                      ],
                    },
                  },
                  {
                    label: 'Pixel 2 (arm7)',
                    seriesConfig: {
                      and: [
                        { missing: 'test' },
                        {
                          prefix: { platform: 'android-hw-p2-8-0-arm7-api-16' },
                        },
                        {
                          eq: {
                            framework: 10,
                            repo: 'mozilla-central',
                            suite: 'raptor-speedometer-geckoview',
                          },
                        },
                      ],
                    },
                  },
                  {
                    label: 'Pixel 2 (ARM64)',
                    seriesConfig: {
                      and: [
                        { missing: 'test' },
                        {
                          eq: {
                            framework: 10,
                            options: 'opt',
                            platform: 'android-hw-p2-8-0-android-aarch64',
                            repo: 'mozilla-central',
                            suite: 'raptor-speedometer-geckoview',
                          },
                        },
                      ],
                    },
                  },
                ]}
              />
            </Grid>
            <Grid item xs={6}>
              <PerfherderGraphContainer
                title="Unity WebGl"
                series={[
                  {
                    label: 'Moto G5 (arm7)',
                    seriesConfig: {
                      and: [
                        { missing: 'test' },
                        {
                          eq: {
                            framework: 10,
                            repo: 'mozilla-central',
                            suite: 'raptor-unity-webgl-geckoview',
                          },
                        },
                        {
                          prefix: { platform: 'android-hw-g5-7-0-arm7-api-16' },
                        },
                      ],
                    },
                  },
                  {
                    label: 'Pixel 2 (arm7)',
                    seriesConfig: {
                      and: [
                        { missing: 'test' },
                        {
                          eq: {
                            framework: 10,
                            repo: 'mozilla-central',
                            suite: 'raptor-unity-webgl-geckoview',
                          },
                        },
                        {
                          prefix: { platform: 'android-hw-p2-8-0-arm7-api-16' },
                        },
                      ],
                    },
                  },
                  {
                    label: 'Pixel 2 (ARM64)',
                    seriesConfig: {
                      and: [
                        { missing: 'test' },
                        {
                          eq: {
                            framework: 10,
                            options: 'opt',
                            platform: 'android-hw-p2-8-0-android-aarch64',
                            repo: 'mozilla-central',
                            suite: 'raptor-unity-webgl-geckoview',
                          },
                        },
                      ],
                    },
                  },
                ]}
              />
            </Grid>
          </Grid>
        </Section>
      </DashboardPage>
    );
  }
}

export default Android;
