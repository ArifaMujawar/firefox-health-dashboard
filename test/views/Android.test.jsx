import React from 'react';
import renderer from 'react-test-renderer';
import Android from '../../src/android';

it('renders correctly', () => {
  const tree = renderer.create(<Android />).toJSON();

  expect(tree).toMatchSnapshot();
});
