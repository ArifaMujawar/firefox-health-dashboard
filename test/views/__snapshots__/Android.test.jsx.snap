// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="DashboardPage-root-1"
>
  <div
    className="DashboardPage-title-2"
  >
    <h1
      style={
        Object {
          "display": "inline",
        }
      }
    >
      Android
      <small
        className="DashboardPage-subtitle-3"
      >
        Release criteria
      </small>
    </h1>
    <div
      style={
        Object {
          "float": "right",
          "padding": "0 1rem 0 0",
        }
      }
    >
      <a
        href="/"
        title="Home"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-4"
          focusable="false"
          role="presentation"
          style={
            Object {
              "margin": "0 .2rem 0 .2rem",
              "verticalAlign": "baseline",
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
          />
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
        </svg>
      </a>
      <a
        href="/android"
        title="Android Performance"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-4"
          focusable="false"
          role="presentation"
          style={
            Object {
              "margin": "0 .2rem 0 .2rem",
              "verticalAlign": "baseline",
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M16 1H8C6.34 1 5 2.34 5 4v16c0 1.66 1.34 3 3 3h8c1.66 0 3-1.34 3-3V4c0-1.66-1.34-3-3-3zm-2 20h-4v-1h4v1zm3.25-3H6.75V4h10.5v14z"
          />
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
        </svg>
      </a>
      <a
        href="/quantum/64"
        title="Quantum 64 Performance"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-4"
          focusable="false"
          role="presentation"
          style={
            Object {
              "margin": "0 .2rem 0 .2rem",
              "verticalAlign": "baseline",
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M21 2H3c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h7v2H8v2h8v-2h-2v-2h7c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm0 14H3V4h18v12z"
          />
        </svg>
      </a>
      <a
        href="/power"
        title="Power Usage"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-4"
          focusable="false"
          role="presentation"
          style={
            Object {
              "margin": "0 .2rem 0 .2rem",
              "verticalAlign": "baseline",
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M15.67 4H14V2h-4v2H8.33C7.6 4 7 4.6 7 5.33v15.33C7 21.4 7.6 22 8.33 22h7.33c.74 0 1.34-.6 1.34-1.33V5.33C17 4.6 16.4 4 15.67 4zM11 20v-5.5H9L13 7v5.5h2L11 20z"
          />
        </svg>
      </a>
      <a
        href="/playback"
        title="Playback"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-4"
          focusable="false"
          role="presentation"
          style={
            Object {
              "margin": "0 .2rem 0 .2rem",
              "verticalAlign": "baseline",
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M17 10.5V7c0-.55-.45-1-1-1H4c-.55 0-1 .45-1 1v10c0 .55.45 1 1 1h12c.55 0 1-.45 1-1v-3.5l4 4v-11l-4 4z"
          />
        </svg>
      </a>
      <a
        href="https://github.com/mozilla-frontend-infra/firefox-health-dashboard/"
        title="Source Code"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-4"
          focusable="false"
          role="presentation"
          style={
            Object {
              "margin": "0 .2rem 0 .2rem",
              "verticalAlign": "baseline",
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M9.4 16.6L4.8 12l4.6-4.6L8 6l-6 6 6 6 1.4-1.4zm5.2 0l4.6-4.6-4.6-4.6L16 6l6 6-6 6-1.4-1.4z"
          />
        </svg>
      </a>
    </div>
  </div>
  <div>
    <div
      className="MuiGrid-container-13 MuiGrid-spacing-xs-24-37"
    >
      <div
        className="MuiGrid-item-14 MuiGrid-grid-xs-6-47"
      >
        <div>
          <div
            className="Section-header-111"
          >
            <h2
              className="Section-title-112"
            >
              <span>
                Bugzilla
              </span>
            </h2>
          </div>
          <div
            className="Section-content-110"
          >
            <div
              className="ChartJsWrapper-chartContainer-114"
            >
              <h2
                className="ChartJsWrapper-title-115"
              >
                <span>
                  Firefox Preview Bugs
                  <a
                    href="https://bugzilla.mozilla.org/buglist.cgi?query_format=advanced&f1=OP&j1=AND&f2=product&o2=equals&v2=GeckoView&f3=OP&j3=OR&f4=resolution&o4=equals&v4=---&f5=resolution&o5=equals&v5=FIXED&f6=CP&f7=status_whiteboard&o7=regexp&v7=.*%5C%5Bgeckoview%3Afenix%3Am.*.*&f8=CP&include_fields=priority%2Ccomponent%2Cassigned_to%2Cbug_status%2Cshort_desc%2Cstatus_whiteboard%2Cchangeddate&order=Bug+Number"
                    rel="noopener noreferrer"
                    target="_blank"
                    title="All GV Firefox Preview MVP bugs"
                  >
                    <div
                      style={
                        Object {
                          "display": "inline-block",
                          "margin": "-1rem 0",
                          "verticalAlign": "middle",
                        }
                      }
                    >
                      <svg
                        aria-hidden="true"
                        className="MuiSvgIcon-root-4"
                        focusable="false"
                        role="presentation"
                        style={
                          Object {
                            "margin": "0 .2rem 0 .2rem",
                            "verticalAlign": "baseline",
                          }
                        }
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M0 0h24v24H0z"
                          fill="none"
                        />
                        <path
                          d="M6 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm12 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-6 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                        />
                      </svg>
                    </div>
                  </a>
                </span>
              </h2>
              <div
                style={
                  Object {
                    "height": 80,
                    "position": "relative",
                  }
                }
              >
                <div
                  style={
                    Object {
                      "position": "absolute",
                      "right": "50%",
                      "top": "50%",
                    }
                  }
                >
                  <div
                    className="MuiCircularProgress-root-117 MuiCircularProgress-colorPrimary-120 MuiCircularProgress-indeterminate-119"
                    role="progressbar"
                    style={
                      Object {
                        "height": 40,
                        "width": 40,
                      }
                    }
                  >
                    <svg
                      className="MuiCircularProgress-svg-122"
                      viewBox="22 22 44 44"
                    >
                      <circle
                        className="MuiCircularProgress-circle-123 MuiCircularProgress-circleIndeterminate-125"
                        cx={44}
                        cy={44}
                        fill="none"
                        r={20.2}
                        strokeWidth={3.6}
                        style={Object {}}
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-item-14 MuiGrid-grid-xs-6-47"
      >
        <div>
          <div
            className="Section-header-111"
          >
            <h2
              className="Section-title-112"
            >
              <span>
                Nimbledroid
                <span
                  className="Section-subtitle-113"
                >
                  Firefox Preview vs Chrome Beta
                </span>
              </span>
            </h2>
          </div>
          <div
            className="Section-content-110"
          >
            <div />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      className="Section-header-111"
    >
      <h2
        className="Section-title-112"
      >
        <span>
          Raptor (TP6m)
          <span>
             
            <a
              href="/android/tp6m?test=cold-loadtime&platform=geckoview-p2-aarch64"
              title="show details"
            >
              <div
                style={
                  Object {
                    "display": "inline-block",
                    "margin": "-1rem 0",
                    "verticalAlign": "middle",
                  }
                }
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root-4"
                  focusable="false"
                  role="presentation"
                  style={
                    Object {
                      "margin": "0 .2rem 0 .2rem",
                      "verticalAlign": "baseline",
                    }
                  }
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M6 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm12 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-6 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                  />
                </svg>
              </div>
            </a>
          </span>
        </span>
      </h2>
    </div>
    <div
      className="Section-content-110"
    >
      <div
        style={
          Object {
            "lineHeight": "100%",
            "textAlign": "center",
            "width": "100%",
          }
        }
      >
        <div
          className="MuiCircularProgress-root-117 MuiCircularProgress-colorPrimary-120 MuiCircularProgress-indeterminate-119"
          role="progressbar"
          style={
            Object {
              "height": 40,
              "width": 40,
            }
          }
        >
          <svg
            className="MuiCircularProgress-svg-122"
            viewBox="22 22 44 44"
          >
            <circle
              className="MuiCircularProgress-circle-123 MuiCircularProgress-circleIndeterminate-125"
              cx={44}
              cy={44}
              fill="none"
              r={20.2}
              strokeWidth={3.6}
              style={Object {}}
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      className="Section-header-111"
    >
      <h2
        className="Section-title-112"
      >
        <span>
          Telemetry
        </span>
      </h2>
    </div>
    <div
      className="Section-content-110"
    >
      <div
        className="MuiGrid-container-13 MuiGrid-spacing-xs-24-37"
      >
        <div
          className="MuiGrid-item-14 MuiGrid-grid-xs-6-47"
        >
          <div>
            <div
              className="ChartJsWrapper-chartContainer-114"
            >
              <h2
                className="ChartJsWrapper-title-115"
              >
                <div>
                  Total content page load time (no 95th)
                   
                  <a
                    href="https://sql.telemetry.mozilla.org/queries/59395"
                    rel="noopener noreferrer"
                    target="_blank"
                    title="go to Redash query"
                  >
                    <div
                      style={
                        Object {
                          "display": "inline-block",
                          "margin": "-1rem 0",
                          "verticalAlign": "middle",
                        }
                      }
                    >
                      <svg
                        aria-hidden="true"
                        className="MuiSvgIcon-root-4"
                        focusable="false"
                        role="presentation"
                        style={
                          Object {
                            "margin": "0 .2rem 0 .2rem",
                            "verticalAlign": "baseline",
                          }
                        }
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M0 0h24v24H0z"
                          fill="none"
                        />
                        <path
                          d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
                        />
                      </svg>
                    </div>
                  </a>
                </div>
              </h2>
              <div
                style={
                  Object {
                    "height": 80,
                    "position": "relative",
                  }
                }
              >
                <div
                  style={
                    Object {
                      "position": "absolute",
                      "right": "50%",
                      "top": "50%",
                    }
                  }
                >
                  <div
                    className="MuiCircularProgress-root-117 MuiCircularProgress-colorPrimary-120 MuiCircularProgress-indeterminate-119"
                    role="progressbar"
                    style={
                      Object {
                        "height": 40,
                        "width": 40,
                      }
                    }
                  >
                    <svg
                      className="MuiCircularProgress-svg-122"
                      viewBox="22 22 44 44"
                    >
                      <circle
                        className="MuiCircularProgress-circle-123 MuiCircularProgress-circleIndeterminate-125"
                        cx={44}
                        cy={44}
                        fill="none"
                        r={20.2}
                        strokeWidth={3.6}
                        style={Object {}}
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-item-14 MuiGrid-grid-xs-6-47"
        >
          <div>
            <div
              className="ChartJsWrapper-chartContainer-114"
            >
              <h2
                className="ChartJsWrapper-title-115"
              >
                <div>
                  Total content page load time
                   
                  <a
                    href="https://sql.telemetry.mozilla.org/queries/59397"
                    rel="noopener noreferrer"
                    target="_blank"
                    title="go to Redash query"
                  >
                    <div
                      style={
                        Object {
                          "display": "inline-block",
                          "margin": "-1rem 0",
                          "verticalAlign": "middle",
                        }
                      }
                    >
                      <svg
                        aria-hidden="true"
                        className="MuiSvgIcon-root-4"
                        focusable="false"
                        role="presentation"
                        style={
                          Object {
                            "margin": "0 .2rem 0 .2rem",
                            "verticalAlign": "baseline",
                          }
                        }
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M0 0h24v24H0z"
                          fill="none"
                        />
                        <path
                          d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
                        />
                      </svg>
                    </div>
                  </a>
                </div>
              </h2>
              <div
                style={
                  Object {
                    "height": 80,
                    "position": "relative",
                  }
                }
              >
                <div
                  style={
                    Object {
                      "position": "absolute",
                      "right": "50%",
                      "top": "50%",
                    }
                  }
                >
                  <div
                    className="MuiCircularProgress-root-117 MuiCircularProgress-colorPrimary-120 MuiCircularProgress-indeterminate-119"
                    role="progressbar"
                    style={
                      Object {
                        "height": 40,
                        "width": 40,
                      }
                    }
                  >
                    <svg
                      className="MuiCircularProgress-svg-122"
                      viewBox="22 22 44 44"
                    >
                      <circle
                        className="MuiCircularProgress-circle-123 MuiCircularProgress-circleIndeterminate-125"
                        cx={44}
                        cy={44}
                        fill="none"
                        r={20.2}
                        strokeWidth={3.6}
                        style={Object {}}
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      className="Section-header-111"
    >
      <h2
        className="Section-title-112"
      >
        <span>
          Perfherder
          <span
            className="Section-subtitle-113"
          >
            Lower in the graph is better regardless if it is a score or execution time (read the Y label)
          </span>
        </span>
      </h2>
    </div>
    <div
      className="Section-content-110"
    >
      <div
        className="MuiGrid-container-13 MuiGrid-spacing-xs-24-37"
      >
        <div
          className="MuiGrid-item-14 MuiGrid-grid-xs-6-47"
        >
          <div
            style={
              Object {
                "position": "relative",
              }
            }
          >
            <div
              className="ChartJsWrapper-chartContainer-114"
            >
              <h2
                className="ChartJsWrapper-title-115"
              >
                <div>
                  Speedometer
                </div>
              </h2>
              <div
                style={
                  Object {
                    "height": 80,
                    "position": "relative",
                  }
                }
              >
                <div
                  style={
                    Object {
                      "position": "absolute",
                      "right": "50%",
                      "top": "50%",
                    }
                  }
                >
                  <div
                    className="MuiCircularProgress-root-117 MuiCircularProgress-colorPrimary-120 MuiCircularProgress-indeterminate-119"
                    role="progressbar"
                    style={
                      Object {
                        "height": 40,
                        "width": 40,
                      }
                    }
                  >
                    <svg
                      className="MuiCircularProgress-svg-122"
                      viewBox="22 22 44 44"
                    >
                      <circle
                        className="MuiCircularProgress-circle-123 MuiCircularProgress-circleIndeterminate-125"
                        cx={44}
                        cy={44}
                        fill="none"
                        r={20.2}
                        strokeWidth={3.6}
                        style={Object {}}
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-item-14 MuiGrid-grid-xs-6-47"
        >
          <div
            style={
              Object {
                "position": "relative",
              }
            }
          >
            <div
              className="ChartJsWrapper-chartContainer-114"
            >
              <h2
                className="ChartJsWrapper-title-115"
              >
                <div>
                  <span>
                    Speedometer CPU power usage
                    <a
                      href="/power?browser=geckoview&suite=speedometer"
                      rel="noopener noreferrer"
                      target="_blank"
                      title="Show details"
                    >
                      <div
                        style={
                          Object {
                            "display": "inline-block",
                            "margin": "-1rem 0",
                            "verticalAlign": "middle",
                          }
                        }
                      >
                        <svg
                          aria-hidden="true"
                          className="MuiSvgIcon-root-4"
                          focusable="false"
                          role="presentation"
                          style={
                            Object {
                              "margin": "0 .2rem 0 .2rem",
                              "verticalAlign": "baseline",
                            }
                          }
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M0 0h24v24H0z"
                            fill="none"
                          />
                          <path
                            d="M6 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm12 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-6 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                          />
                        </svg>
                      </div>
                    </a>
                  </span>
                </div>
              </h2>
              <div
                style={
                  Object {
                    "height": 80,
                    "position": "relative",
                  }
                }
              >
                <div
                  style={
                    Object {
                      "position": "absolute",
                      "right": "50%",
                      "top": "50%",
                    }
                  }
                >
                  <div
                    className="MuiCircularProgress-root-117 MuiCircularProgress-colorPrimary-120 MuiCircularProgress-indeterminate-119"
                    role="progressbar"
                    style={
                      Object {
                        "height": 40,
                        "width": 40,
                      }
                    }
                  >
                    <svg
                      className="MuiCircularProgress-svg-122"
                      viewBox="22 22 44 44"
                    >
                      <circle
                        className="MuiCircularProgress-circle-123 MuiCircularProgress-circleIndeterminate-125"
                        cx={44}
                        cy={44}
                        fill="none"
                        r={20.2}
                        strokeWidth={3.6}
                        style={Object {}}
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-item-14 MuiGrid-grid-xs-6-47"
        >
          <div
            style={
              Object {
                "position": "relative",
              }
            }
          >
            <div
              className="ChartJsWrapper-chartContainer-114"
            >
              <h2
                className="ChartJsWrapper-title-115"
              >
                <div>
                  Unity WebGl
                </div>
              </h2>
              <div
                style={
                  Object {
                    "height": 80,
                    "position": "relative",
                  }
                }
              >
                <div
                  style={
                    Object {
                      "position": "absolute",
                      "right": "50%",
                      "top": "50%",
                    }
                  }
                >
                  <div
                    className="MuiCircularProgress-root-117 MuiCircularProgress-colorPrimary-120 MuiCircularProgress-indeterminate-119"
                    role="progressbar"
                    style={
                      Object {
                        "height": 40,
                        "width": 40,
                      }
                    }
                  >
                    <svg
                      className="MuiCircularProgress-svg-122"
                      viewBox="22 22 44 44"
                    >
                      <circle
                        className="MuiCircularProgress-circle-123 MuiCircularProgress-circleIndeterminate-125"
                        cx={44}
                        cy={44}
                        fill="none"
                        r={20.2}
                        strokeWidth={3.6}
                        style={Object {}}
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
